src/Twig/Component/WebElements/SummaryPaymentComponent.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FluxSE\SyliusStripePlugin\Twig\Component\WebElements;
+
+use FluxSE\SyliusStripePlugin\Exception\WebElementsSummaryException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Sylius\Bundle\UiBundle\Twig\Component\ResourceLivePropTrait;
+use Sylius\Bundle\UiBundle\Twig\Component\TemplatePropTrait;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Sylius\Resource\Model\ResourceInterface;
+use Sylius\TwigHooks\LiveComponent\HookableLiveComponentTrait;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+
+#[AsLiveComponent()]
+class SummaryPaymentComponent
+{
+    use DefaultActionTrait;
+    use HookableLiveComponentTrait;
+    use ResourceLivePropTrait;
+    use TemplatePropTrait;
+
+    #[LiveProp(hydrateWith: 'hydrateResource', dehydrateWith: 'dehydrateResource')]
+    #[ExposeInTemplate('cart')]
+    public ?ResourceInterface $cart = null;
+
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly PaymentRequestRepositoryInterface $paymentRequestRepository,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+    }
+
+    public function mount(): void
+    {
+        try {
+            $this->cart = $this->getOrderFromPaymentRequest();
+        } catch (WebElementsSummaryException $e) {
+            $this->logger->error(sprintf('[Payment Summary] Error: %s', $e->getMessage()));
+            throw $e;
+        }
```

Comment:
I understand why this can be needed but to separate concerns you can keep this try catch in a decorated service on your own project.
This live component must only do one thing "retrieve the current payment request". Then developers can decorate it and enhance it with whatever they want.